tests/Integration/Behaviour/Features/Context/Domain/CurrencyFeatureContext.php

```diff
@@ -0,0 +1,334 @@
+        } catch (CoreException $e) {
+            $this->lastException = $e;
+        }
+    }
+
+    /**
+     * @When I edit currency :reference with following properties:
+     */
+    public function editCurrency($reference, TableNode $node)
+    {
+        $data = $node->getRowsHash();
+        /** @var Currency $currency */
+        $currency = SharedStorage::getStorage()->get($reference);
+
+        $command = new EditCurrencyCommand((int) $currency->id);
+
+        if (isset($data['iso_code'])) {
+            $command->setIsoCode($data['iso_code']);
+        }
+
+        if (isset($data['exchange_rate'])) {
+            $command->setExchangeRate((float) $data['exchange_rate']);
+        }
+
+        if (isset($data['is_enabled'])) {
+            $command->setIsEnabled((bool) $data['is_enabled']);
+        }
+
+        if (isset($data['shop_association'])) {
+            $command->setShopIds([(int) $data['shop_association']]);
+        }
+
+        try {
+            $this->lastException = null;
+            $this->getCommandBus()->handle($command);
+
+            SharedStorage::getStorage()->set($reference, new Currency($currency->id));
+        } catch (CoreException $e) {
+            $this->lastException = $e;
+        }
+    }
+
+    /**
+     * @When I disable currency ":currencyReference"
+     */
+    public function disableCurrency($reference)
+    {
+        /** @var Currency $currency */
+        $currency = SharedStorage::getStorage()->get($reference);
+
+        try {
+            $this->lastException = null;
+            $this->getCommandBus()->handle(new ToggleCurrencyStatusCommand((int) $currency->id));
+        } catch (CannotDisableDefaultCurrencyException $e) {
+            $this->lastException = $e;
+        }
+    }
+
+    /**
+     * @When I delete currency ":currencyReference"
+     */
+    public function deleteCurrency($reference)
+    {
+        /** @var Currency $currency */
+        $currency = SharedStorage::getStorage()->get($reference);
+
+        try {
+            $this->lastException = null;
+            $this->getCommandBus()->handle(new DeleteCurrencyCommand((int) $currency->id));
+        } catch (CannotDeleteDefaultCurrencyException $e) {
+            $this->lastException = $e;
+        }
+    }
+
+    /**
+     * @Then currency :reference should be :isoCode
+     */
+    public function assertCurrencyIsoCode($reference, $isoCode)
+    {
+        /** @var Currency $currency */
+        $currency = SharedStorage::getStorage()->get($reference);
+
+        if ($currency->iso_code !== $isoCode) {
+            throw new RuntimeException(sprintf(
+                'Currency "%s" has "%s" iso code, but "%s" was expected.',
+                $reference,
+                $currency->iso_code,
+                $isoCode
+            ));
+        }
+    }
+
+    /**
+     * @Then /^currency "(.*)" should have status (enabled|disabled)$/
```

Comment:
For non greedy.
```suggestion
     * @Then /^currency "(.*?)" should have status (enabled|disabled)$/
```

Reply:
I don't think it's really needed here thanks to the enclosing `""` Besides all conditions are written that way for now

Reply:
No prob, hope we don't add a `@Then` statement in the future which will come into conflict :)

Reply:
Sure, we can always change the regex when that occurs